rehfeldmedical/metis_umbrella
# Run tests in CI

This adds test running as part of CI for pull requests (except those with `nodeploy`).

TODO: Add test running for `develop`/`master` branches too

### Changes
- Add elixir container to build pod used by Jenkinsfile
- Spin up test-only ephemeral PostgreSQL to run tests against
- Add deployment-config/postgresql-ephemeral.yaml so you can set admin
password in container
- Add junit-formatter to apps/metis to output JUnit-style xml from tests
- Collect junit test results after running tests

### To test
1. Run `MIX_ENV=test POSTGRESQL_DATABASE=test mix test` locally
2. Trigger a CI build on OpenShift Jenkins and observe that: (i) it runs tests, (ii) JUnit-style test results are collected under the build in Jenkins
